frog/imports/ui/StudentView/Runner.jsx

```diff
@@ -108,23 +110,54 @@ type PropsT = {
   username: string,
   userid: string,
   stream: Function,
+  groupingKey: string,
   groupingValue: string,
   activityTypeId: string,
   readOnly?: boolean,
-  sessionId: string
+  sessionId: string,
+  activityId: string
 };
 
 export class RunActivity extends React.Component<PropsT, {}> {
   ActivityToRun: any;
 
   constructor(props: PropsT) {
     super();
-    const { reactiveId, activityTypeId, readOnly } = props;
+    const {
+      reactiveId,
+      activityTypeId,
+      readOnly,
+      activityId,
+      userid,
+      sessionId,
+      groupingKey,
+      groupingValue
+    } = props;
     const activityType = activityTypesObj[activityTypeId];
+    const meta: {
+      createdInActivity: string,
+      createdByUser: string,
+      createdByInstance?: Object,
+      sessionId: string
+    } = {
+      createdInActivity: activityId,
+      createdByUser: userid,
+      sessionId
+    };
+    if (groupingKey) {
+      meta.createdByInstance = { [groupingKey]: groupingValue };
```

Comment:
Is that the value we want here? I think that having `createdByInstance: groupingValue` which will be `userId` or `all` in P1 and P3 is much easier to use in general. Then we can also carry along `meta.groupingKey` which will be undefined for P1 and P3 and tell us what was the grouping based on in P2